packages/analytics-browser/src/snippet-index.ts

```diff
@@ -1,18 +1,28 @@
+import { getGlobalScope } from '@amplitude/analytics-client-common';
 import * as amplitude from './index';
 import { runQueuedFunctions } from './utils/snippet-helper';
-import { GlobalScope } from './utils/global-scope';
 
-GlobalScope.amplitude = Object.assign(GlobalScope.amplitude || {}, amplitude);
+// https://developer.mozilla.org/en-US/docs/Glossary/IIFE
```

Comment:
Curious to learn why we use IIFE here? Is it to isolate the scope?

Reply:
the reason is very shallow 😅  only because i wanted to do 

```
if () {
  ...
  return;
}

...
```

than

```
if () {
  ...
} else {
  ...
}
```

we can only call `return` in function level, and not in top level

Reply:
Got it, thanks for sharing!